src/site/topic23.rst

```diff
@@ -83,6 +83,61 @@ Naive Search
 Binary Search
 -------------
 
+.. image:: img/binarysearchtree_binarysearch0.png
+   :width: 250 px
+   :align: center
+
+* Consider the above binary search tree
+* Based on the special ordering binary search trees have, can you devise an improved search?
+    * **Hint:** If I am looking for a number greater than 14, which subtree must the element be in if it exists in the tree?
+
+.. code-block:: java
+    :linenos:
+
+    public boolean contains(T needle) {
+        return binarySearch(root, needle) != null;
+    }
+
+    private boolean binaryContains(Node<T> current, T needle) {
+        if (current == null) {
+            return false;
+        } else if (current.getData().equals(needle)) {
+            return true;
+        } else {
+            if (current.getData().compareTo(needle) > 0) {
+                return binarySearch(current.getLeft(), needle);
+            } else {
+                return binarySearch(current.getRight(), needle);
+            }
+        }
+    }
+
+* Notice the similarity between the naive search and the binary search
+* They are nearly the same, except you will notice that, the naive search may search both subtrees
+* However, binary search will only ever travel down one subtree due to the ``if (current.getData().compareTo(needle) > 0) `` statement
+
+.. image:: img/binarysearchtree_binarysearch1.png
+   :width: 250 px
+   :align: center
+
+* In the above example, if we search for the number 13, we can conclude that it must not exist in the tree since
+    * 13 is less than 14, so it must be in the left subtree of the node containing 14
+    * 13 is greater than 8, thus it must be in the right subtree
+    * 13 is greater than 12, therefore it must be in the right subtree
+    * 12 has no right subtree, therefore 13 must **not** be in the tree
```

Comment:
> Think I should change the first example to searching for 12 instead? That way it shows a success instead of two false examples.

It definitely feels weird showing only failure cases. This skips one of the recursive base cases, the thing being in the tree. I'd either:
* do `12`, showing it finds the element + do `22`, showing it does not find the element
* do `12` + say, _similarly, `13` must **not** be in the tree since blah_; and then do `23` and `22`
* do `14` (skim over it), then `12`, then `13` or `10`